uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -167,6 +194,37 @@ define([
         },
 
         /**
+         * Sets the flag to mark file for deletion. The file is not yet deleted
+         * from the server.
+         */
+        deleteAttachment: function (attachmentID) {
+          this.files = this.files.map(function (file) {
+            if (file.attachment_id == attachmentID && !file.toBeDeleted) {
+              file.toBeDeleted = true;
+            }
+
+            return file;
+          });
+        },
+
+        /**
+         * Removes a comment from memory
+         *
+         * @param {Object} commentObj - comment object
+         */
+        removeComment: function (commentObj) {
+          //If its an already saved comment, mark a toBeDeleted flag
+          if (commentObj.comment_id) {
+            commentObj.toBeDeleted = true;
+            return;
+          }
+
+          this.comments = _.reject(this.comments, function (comment) {
```

Comment:
Another difference in approach between comment and attachments,
For comments
1. Object is removed from array if it does not have an ID

For attachements
1. Object is only marked as deleted but not removed from array. 

Please keep similar approach in both. 
Comments is already tested, so keep that?